Update introduction to solutions (#3005)
elastic/docs-content: get-started/introduction.md

---
mapped_pages:
  - https://www.elastic.co/guide/en/elasticsearch/reference/current/elasticsearch-intro-what-is-es.html
navigation_title: Solutions
products:
  - id: elasticsearch
applies_to:
  stack:
  serverless:
navigation_title: Solutions
---

# Solutions overview [introduction]

Elastic offers three major search-powered solutions: {{es}}, Elastic {{observability}}, and {{elastic-sec}}—all built on an open source, extensible platform.
Whether you're building a search experience, monitoring your infrastructure, or securing your environment, there is a solution that is right for your business needs.

| Your need | Recommended solution | Best for |
|-----------|-------------------|----------|
| Build powerful, scalable searches to quickly search, analyze, and visualize large amounts of data for real-time insights| [{{es}}](/solutions/search.md) - [Get started](/solutions/search/get-started.md)| Developers, architects, data engineers |
| Observe and monitor system health and performance, or send telemetry data | [Elastic {{observability}}](/solutions/observability.md) - [Get started](/solutions/observability/get-started.md) | DevOps, SREs, IT operations |

> | Monitor data for anomalous activity, detect, prevent, and respond to security incidents | [{{elastic-sec}}](/solutions/security.md) -  [Get started](/solutions/security/get-started.md)| SOC teams, security analysts, IT security admins |

Reviewer: ```suggestion
| Build powerful, scalable searches to quickly search, analyze, and visualize large amounts of data for real-time insights| • [{{es}}](/solutions/search.md)<br>• [Get started](/solutions/search/get-started.md)| Developers, architects, data engineers |
| Observe and monitor system health and performance, or send telemetry data | • [Elastic {{observability}}](/solutions/observability.md)<br>• [Get started](/solutions/observability/get-started.md) | DevOps, SREs, IT operations |
| Monitor data for anomalous activity, detect, prevent, and respond to security incidents | • [{{elastic-sec}}](/solutions/security.md)<br>• [Get started](/solutions/security/get-started.md)| SOC teams, security analysts, IT security admins |
```
Nit: a trick to make it clearer there are actually 2 links every time

| Your need | Recommended solution | Best for |
|-----------|-------------------|----------|
| Build powerful, scalable searches to quickly search, analyze, and visualize large amounts of data for real-time insights  | [{{es}}](#search-overview) | Developers, architects, data engineers |
| Observe and monitor system health and performance, or send telemetry data | [Elastic {{observability}}](#observability-overview) | DevOps, SREs, IT operations |
| Monitor data for anomalous activity, detect, prevent, and respond to security incidents | [{{elastic-sec}}](#security-overview) | SOC teams, security analysts, IT security admins |

:::{tip}
Check out our [customer success stories](https://www.elastic.co/customers/success-stories) to learn how various organizations utilize our products for their specific business needs.
:::

Each of our solutions is available as a fully managed {{serverless-short}} project or a self-managed deployment. Refer to [deployment options](../get-started/deployment-options.md) to learn about these options. 

## {{es}} [search-overview]

> {{es-serverless}} projects and the {{es}} solution combine the core {{es}} data store, search engine, and vector database technologies with specialized user interfaces and tools, giving you the building blocks to create, deploy, and run your own search applications.

Reviewer 2: I really liked how you described serverless and self-managed in the previous paragraph. I think because you set that up so nicely, you could just say Elasticsearch here. 
```suggestion
{{es}} combines the core {{es}} data store, search engine, and vector database technologies with specialized user interfaces and tools, giving you the building blocks to create, deploy, and run your own search applications.
```

<!--
### {{es}} use cases [search-use-cases]
-->
For an overview of search use cases, including full-text, geospatial, semantic, and vector search, as well as retrieval augmented generation (RAG), go to [](/solutions/search.md).
To try out some simple search use cases, go to [](/solutions/search/get-started/quickstarts.md).
<!--
### {{es}} core concepts [search-concepts]
-->
For an introduction to core {{es}} concepts such as indices, documents, and mappings, refer to [](/manage-data/data-store.md).

> To dive more deeply into the building blocks of an {{es}} cluster, including nodes, shards, primaries, and replicas, refer to [](/deploy-manage/distributed-architecture.md).

Reviewer 2: ```suggestion
To dive more deeply into the building blocks of {{es}} clusters, including nodes, shards, primaries, and replicas, refer to [](/deploy-manage/distributed-architecture.md).
```

<!--
The following concepts are not unique to this solution.
 ::::{dropdown} Concepts

The following concepts relate to core {{es}} features and thus apply to all solutions:

> * **Index:** A collection of documents with similar characteristics that are uniquely identified by a name or an alias. The name is used to target the index in search queries and other operations.

Reviewer 2: ```suggestion
* **Index:** A collection of related documents that are uniquely identified by a name or an alias. The name is used to target the index in search queries and other operations.
```

> * **Document:** Any structured data encoded in JSON. {{es}} organizes and stores data into documents.

Reviewer 2: ```suggestion
* **Document:** Any structured data encoded in JSON. {{es}} organizes and stores documents.
```

* **Field:** The smallest individual unit of data within a document. It represents a specific property or attribute of the data you're indexing (for example, title, author, date, or summary). Fields are critical for indexing, as they determine how data is analyzed and stored to enable efficient searching.
* **Mapping:** The process that defines how a document and its fields are stored and indexed.
* **Client:** Software or an application that facilitates communication and interaction with an {{es}} cluster. It enables applications written in various programming languages to send requests to {{es}}, process the response, and then push that data into the cluster.

> * **Primary shard:** A self-contained Lucene index that contains some or all data for an index. Shards allow {{es}} to scale horizontally by splitting an index's data into smaller, manageable partitions, improving performance. Each document in an index belongs to one primary shard.

Reviewer 2: ```suggestion
* **Primary shard:** A self-contained Lucene index that contains some or all of an index's data. Shards allow {{es}} to scale horizontally by splitting an index's data into smaller, manageable partitions, improving performance. Each document in an index belongs to one primary shard.
```

* **Digital Experience Monitoring:**
  * **Real User Monitoring (RUM):** Capture and analyze data on how real users interact with web applications to improve perceived performance.
  * **Synthetic Monitoring:** Proactively simulate user journeys and API calls to test application availability and functionality.
  * **Uptime Monitoring:** Continuously check the status of services and applications to ensure they are available.
* **Universal Profiling:** Gain visibility into system performance and identify expensive lines of code without application instrumentation, helping to increase CPU efficiency and reduce cloud spend.
* **LLM Observability:** Gain deep insights into the performance, usage, and costs of Large Language Model (LLM) prompts and responses.
* **Incident Response and Management:** Facilitate the investigation of operational incidents by correlating data from multiple sources, which accelerates root cause analysis and resolution.
:::

### {{observability}} core concepts [observability-concepts]

> At the heart of Elastic {{observability}} are several key concepts that enable its capabilities. 

Reviewer 2: ```suggestion
At the heart of Elastic {{observability}} are several key components that enable its capabilities. 
```

:::{dropdown} Concepts
* The three pillars of {{observability}} are: 
  * [**Logs:**](/solutions/observability/logs.md) Timestamped records of events that provide detailed, contextual information.
  * [**Metrics:**](/solutions/observability/infra-and-hosts/analyze-infrastructure-host-metrics.md) Numerical measurements of system performance and health over time.

>   * [**Traces:**](/solutions/observability/apm/traces.md) A representation of the end-to-end journey of a request as it travels through a distributed system.

Reviewer 2: ```suggestion
  * [**Traces:**](/solutions/observability/apm/traces.md) Representations of end-to-end journeys of requests as they travel through distributed systems.
```
trying to match the two previous bullet points

* [**OpenTelemetry:**](/solutions/observability/apm/use-opentelemetry-with-apm.md) {{Observability}} offers first-class, production-grade support for OpenTelemetry. This allows organizations to use vendor-neutral instrumentation and stream native OTel data without proprietary agents, leveraging the Elastic Distribution of OpenTelemetry (EDOT).
* [**AIOps and AI Assistant:**](/solutions/observability/observability-ai-assistant.md) Leverages predictive analytics and an LLM-powered AI Assistant to reduce the time required to detect, investigate, and resolve incidents. This includes zero-config anomaly detection, pattern analysis, and the ability to surface correlations and root causes.

> * **[Alerting](/solutions/observability/incident-management/alerting.md) and [Cases](/solutions/observability/incident-management/cases.md):** A built-in feature for creating rules to detect complex conditions and trigger actions. It allows teams to stay aware of potential issues and use cases to track investigation details, assign tasks, and collaborate on resolutions.

Reviewer 2: Bit confusing here how the bullet lists both Alerting and Cases (plural) and then starts with "_a_ built-in feature" (singular)

* [**Service Level Objectives (SLOs):**](/solutions/observability/incident-management/service-level-objectives-slos.md) A framework for defining and monitoring the reliability of a service. Elastic {{observability}} allows for creating and tracking SLOs to ensure that performance targets are being met.
:::

## {{elastic-sec}}

### Security overview [security-overview]

> {{elastic-sec}} is a unified security solution that integrates SIEM (Security Information and Event Management), endpoint security, and cloud security into a single platform so you can detect, prevent, and respond to cyber threats across your entire environment in near real time. Elastic Security leverages {{es}}'s powerful platform for its searching and analytic capabilities, and {{kib}} for its visualization features. By combining prevention, detection, and response capabilities, {{elastic-sec}} helps your organization reduce the risk of successful attacks. 

Reviewer 2: ```suggestion
{{elastic-sec}} is a unified security solution that integrates SIEM (Security Information and Event Management), endpoint security, and cloud security into a single platform so you can detect, prevent, and respond to cyber threats across your entire environment in near real time. Elastic Security leverages {{es}}'s powerful search and analytics capabilities, and {{kib}}'s visualization and collaboration features. By combining prevention, detection, and response capabilities, {{elastic-sec}} helps your organization reduce its security risk. 
```
It's fine as is, but here's another option. I thought "reduce the risk of successful attacks" didn't sound very inspiring, but maybe it doesn't need to.

> Install {{elastic-sec}} on one of our Elastic-managed Cloud deployments or your own self-managed infrastructure.  

Reviewer 2: maybe a good place to use the {{ecloud}} attribute?

### Security use cases [security-use-cases]

> Use {{elastic-sec}} to protect your systems from the latest threats.

Reviewer 2: ```suggestion
Use {{elastic-sec}} to protect your systems from security threats.
```
Or maybe "a wide range of threats"? It's not just the latest threats. Old ones too.

:::{dropdown} Use cases

> * **SIEM:** {{elastic-sec}} is a modern SIEM that provides a centralized platform for ingesting, analyzing, and managing security data from various sources. 

Reviewer 2: ```suggestion
* **SIEM:** {{elastic-sec}}'s modern SIEM provides a centralized platform for ingesting, analyzing, and managing security data from various sources. 
```

* **Third-party integration support:** Ingest data from a various tools and data sources so you can centralize your security data.
* **Threat detection and analytics:** Identify unknown threats by enabling prebuilt or custom detection rules, automatically detect anomalous activity with built-in machine learning jobs, or proactively search for threats using our powerful threat hunting and interactive visualization tools. 
* **Automatic migration:** Migrate SIEM rules from other platforms to {{elastic-sec}}. 
* **Endpoint protection and threat prevention:** Automatically stop cybersecurity attacks—such as malware and ransomware—before damage and loss can occur.

> * **AI-powered features:** Elastic Security leverages generative AI to help enhance threat detection, assist with incident response, and day-to-day security operations. For example, the AI Assistant can summarize alerts, identify relevant information, suggest investigation steps, and generate complex queries from natural language input.

Reviewer 2: ```suggestion
* **AI-powered features:** Leverage generative AI to help enhance threat detection, assist with incident response, and improve day-to-day security operations. For example, use AI Assistant to summarize alerts, identify relevant information, suggest investigation steps, and generate complex queries from natural language input.
```

* **Custom dashboards and visualizations:** Create custom dashboards and visualizations to gain insights into security events.
* **Cloud Security:** {{elastic-sec}} provides the following cloud features:
  * **Cloud Security Posture Management (CSPM) and Kubernetes Security Posture Management (KSPM):** Check cloud service configurations against security benchmarks to identify and resolve misconfigurations that can be exploited.
  * **Cloud Workload Protection:** Get visibility and runtime protection for cloud workloads.
  * **Vulnerability Management:** Uncover vulnerabilities within your cloud infrastructure.
:::

### Security core concepts [security-concepts]

Before diving into setup and configuration, familiarize yourself with the foundational terms and core concepts that power {{elastic-sec}}. 

:::{dropdown} Concepts 

> * **{{agent}}:** A single, unified way to add monitoring for logs, metrics, and other types of data to a host. Elastic Agent can also protect hosts from security threats, query data from operating systems, and forward data from remote services or hardware. 

Reviewer 2: ```suggestion
* **{{agent}}:** A single, unified way to collect logs, metrics, and other types of data from a host. Elastic Agent can also protect hosts from security threats, query data from operating systems, and forward data from remote services or hardware. 
```

* **{{elastic-defend}}:** {{elastic-sec}}'s Endpoint Detection and Response (EDR) tool that protects endpoints from malicious activity. {{elastic-defend}} uses a combination of techniques like machine learning, behavioral analysis, and prebuilt rules to detect, prevent, and respond to threats in real-time.

> * **{{elastic-endpoint}}:** The installed component that performs {{elastic-defend}}'s threat monitoring and prevention capabilities. 

Reviewer 2: ```suggestion
* **{{elastic-endpoint}}:** The software component that performs {{elastic-defend}}'s threat monitoring and prevention capabilities. 
```

Love how this describes the relationship between endpoint and defend. Can we say something about the relationship between Elastic Endpoint and Elastic Agent?

> * **Detection engine:** The framework that detects threats by using rules to search for suspicious events in data sources and generating alerts when those rules meet the defined criteria.  

Reviewer 2: ```suggestion
* **Detection engine:** The framework that detects threats by using rules to search for suspicious events in your data and generates alerts when events meet a rule's criteria.  
```

> * **Detection rules:** Sets of conditions that identify potential threats and malicious activities. Rules analyze various data sources, including logs and network traffic, to detect anomalies, suspicious behaviors, or known attack patterns. {{elastic-sec}} ships out-of-the-box prebuilt rules, or you can create your own custom rules. 

Reviewer 2: ```suggestion
* **Detection rules:** Sets of conditions that identify potential threats and malicious activities. Rules analyze various data sources, including logs and network traffic, to detect anomalies, suspicious behaviors, or known attack patterns. {{elastic-sec}} ships out-of-the-box prebuilt rules, and you can create your own custom rules. 
```

> * **Alerts:** A notification that's generated when a rule’s criteria are met. You can then investigate an alert to dive into deeper details.  

Reviewer 2: ```suggestion
* **Alerts:** Notifications that are generated when rule conditions are met. Alerts include a wide range of information about potential threats, including host, user, network, and other contextual data to assist your investigation.
```

* **Machine learning and anomaly detection:** Anomaly detection jobs identify anomalous events or patterns in your data. Use these with machine learning detection rules to generate alerts when behavior deviates from normal activity.

> * **Entity analytics:** A threat detection feature that combines the power of Elastic’s detection engine and machine learning capabilities to identify unusual user behaviors across hosts, users, and services. Entity analytics uses a risk scoring engine to calculate a risk score, which is evaluated at a recurring interval. 

Reviewer 2: ```suggestion
* **Entity analytics:** A threat detection feature that combines the power of Elastic’s detection engine and machine learning capabilities to identify unusual behavior for hosts, users, and services. 
```
Second sentence felt like at the same too much detail, and not enough detail (I'm left with questions that are probably out of scope for this page).

* **Cases:** A tool that allows you to collect and share information about security issues. Opening a case lets you track key investigation details and collect alerts in a central location. You can also send cases to external systems.
* **Timeline:** A threat hunting tool that allows you to investigate security events so you can gather and analyze data related to alerts or suspicious activity. You can add events to Timeline from various sources, build custom queries, and import/export a Timeline to collaborate and share. 

> * **Security posture management:** Includes two Cloud Security features–Cloud Security Posture Management (CSPM) and Kubernetes Security Posture Management (KSPM)–that help you evaluate the services and resources in your cloud environment, such as storage, compute, IAM, and more—against security guidelines defined by the Center for Internet Security (CIS). These features help you identify and remediate configuration risks in your environment. 

Reviewer 2: ```suggestion
* **Security posture management:** Includes native Cloud Security features such as Cloud Security Posture Management (CSPM) and Cloud Native Vulnerability Management (CNVM)–that help you evaluate your cloud infrastructure's configuration against security best practices and identify vulnerabilities. You can use Elastic's native tools or ingest third-party cloud security data and incorporate it into Elastic Security's workflows.
```

Reviewer 2: Conceptually I think it makes sense to de-emphasize KSPM, because yes we have it but it's not in active development. CNVM to me fits better alongside CSPM. I also think it's good to mention our third-party integrations here.

* **AI Assistant:** A generative AI-powered tool that helps with tasks like alert investigation, incident response, and query generation. It utilizes natural language processing and knowledge retrieval to provide context-aware assistance, summarize threats, suggest next steps, and automate workflows. Use AI Assistant to better understand and respond to security incidents.
:::

<!--TBD: Call out how solutions map to Serverless project types? -->
<!-- Content moved from the-stack.md
## Get data into Elasticsearch

When building custom search experiences or indexing general data, you have several direct and flexible ingestion options:

* **Native APIs and language clients:** Index any JSON document directly using the {{es}} REST API or the official clients for languages like Python, Java, Go, and more.  

> * **Web crawler:** Ingest content from public or private websites to make them searchable.  

Reviewer 2: ```suggestion
* **Web crawler:** Ingest content from public or private websites to make it searchable.  
```

* **Enterprise connectors:** Use pre-built connectors to sync data from external content sources like SharePoint, Confluence, Jira, and databases like MongoDB or PostgreSQL into {{es}}.

## Get data into Elastic Observability

For full-stack observability, ingest logs, metrics, traces, and profiles using these OpenTelemetry-native methods:

> * **{{edot}}:** Use Elastic's supported OpenTelemetry SDKs for custom application instrumentation and the Collector for vendor-neutral infrastructure telemetry.  

Reviewer 2: ```suggestion
* **{{edot}}:** Use Elastic's OpenTelemetry SDKs for custom application instrumentation and the Collector for vendor-neutral infrastructure telemetry.  
```

> * **{{agent}}:** A single agent to collect infrastructure logs and metrics from hosts, containers, and cloud services using pre-built integrations.  

Reviewer 2: ```suggestion
* **{{agent}}:** Collect infrastructure logs and metrics from hosts, containers, and cloud services using pre-built integrations.  
```
Starting with a verb like the other bullets
